### DISCUSSION TITLE
导出[B站 - 记笔记]功能到本地或其它生产力工具

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1583] [服务]
> 此适配规则由来自社区的 @binsee 制作

***

## 手动升级官方适配列表

**注意：实际只要大于下图的值即可。**

<img width="1553" alt="Xnip2020-12-22_11-23-25" src="https://user-images.githubusercontent.com/81074/102845268-268c6000-4448-11eb-9521-c5414379a602.png">

## 使用说明

- 务必将 `https://space.bilibili.com/**` 加入 **排除列表**（下图所示），然后在笔记页面中点击要导出的笔记后，手动AA进入阅读模式
- 由于笔记图片是通过cookie来从服务器获取，因此无法作为外链使用。要在其他地方使用图片请导出为离线html
- 导出的pdf会无法显示图片，可先导出为 **离线html**，打开后再打印为pdf文档

<img width="1481" alt="Xnip2020-12-22_11-28-34" src="https://user-images.githubusercontent.com/81074/102845610-e24d8f80-4448-11eb-8be6-31000b88262f.png">

## 适配特性

- 阅读模式开头增加视频链接
- 笔记中的视频时间点标记将标记为链接，点击后打开视频对应时间点

## 显示效果

- 哔哩哔哩笔记页面显示效果
![image](https://user-images.githubusercontent.com/5285894/102804958-75f57080-43f5-11eb-8546-b42a2a2c8001.png)

- 导出为离线html显示效果
![image](https://user-images.githubusercontent.com/5285894/102806053-257f1280-43f7-11eb-8866-8114e7b40548.png)

- 导出为md，使用Typora打开
![image](https://user-images.githubusercontent.com/5285894/102807960-93790900-43fa-11eb-9df3-9b24505baa0d.png)

## 关联

如果使用 assets + md 格式时无法显示图片，可根据 [此方式](https://github.com/Kenshin/simpread/issues/4510#issuecomment-1323116938) 解决。

**Comment by heyhuihui1** [comment]
导出成md之后为什么图片没有办法显示了呢？

**Reply by Kenshin** [reply]
嗨，给个 URL 瞧瞧

**Reply by binsee** [reply]
> 导出成md之后为什么图片没有办法显示了呢？

因为b站笔记中的图片地址是非公开地址，需要用户权鉴。其image_id是指向此用户笔记系统中图片列表的索引号

**Reply by Kenshin** [reply]
@binsee 

离线 HTML 是不是就没问题了？

**Reply by binsee** [reply]
是的。
- `导出为 离线html` 和 `导出为 离线markdown` 会将图片以base64编码形式写入到文件内，因此可以正常显示。
- `导出为 html` 和 `导出为 markdown` 内图片仍然是b站的图片地址，渲染显示不会传递cookie，因此不会显示。

**Comment by JokAlec** [comment]
我记得前上个月还使用着没问题，现在按AA大部分笔记会丢失，导出markdown视频时间点的链接也木有了

**Reply by binsee** [reply]
确认是由于b站对笔记功能进行了修改，只需要修改适配规则的url为`http*://space.bilibili.com/v/note-list`即可，稍后发pr

**Reply by Kenshin** [reply]
@JokAlec 

新的适配规则已发布了，请 [手动更新](http://ksria.com/simpread/docs/#/%E7%AB%99%E7%82%B9%E9%80%82%E9%85%8D%E6%BA%90?id=%e5%ae%98%e6%96%b9%e9%80%82%e9%85%8d%e6%ba%90) 即可。

**Reply by JokAlec** [reply]
哎妈，效率太高了

**Reply by Kenshin** [reply]
@JokAlec 

主要是 @binsee 大佬高效！

**Reply by chinyajie** [reply]
> 确认是由于b站对笔记功能进行了修改，只需要修改适配规则的url为`http*://space.bilibili.com/v/note-list`即可，稍后发pr

您好，请问这个新地址要填写在哪里呢？

**Reply by binsee** [reply]
> > 确认是由于b站对笔记功能进行了修改，只需要修改适配规则的url为`http*://space.bilibili.com/v/note-list`即可，稍后发pr
> 
> 您好，请问这个新地址要填写在哪里呢？

b站笔记居然又更新了，稍后发pr，请等待官方规则适配更新。
如果想提前使用，可将此地址导入到『第三方适配源』： https://gist.githubusercontent.com/binsee/55acad05b921ba79ac6695a9463f563c/raw/note.space.bilibili.com.json

**Reply by chinyajie** [reply]
> > > 确认是由于b站对笔记功能进行了修改，只需要修改适配规则的url为`http*://space.bilibili.com/v/note-list`即可，稍后发pr
> > 
> > 
> > 您好，请问这个新地址要填写在哪里呢？
> 
> b站笔记居然又更新了，稍后发pr，请等待官方规则适配更新。
> 如果想提前使用，可将此地址导入到『第三方适配源』： https://gist.githubusercontent.com/binsee/55acad05b921ba79ac6695a9463f563c/raw/note.space.bilibili.com.json

谢谢👍🏻

**Comment by okerivy** [comment]
1，为啥我的没有 上面那个说的 排除列表？ 
<img src="https://user-images.githubusercontent.com/4578474/116810850-9d650280-ab78-11eb-8f6a-f55fe56898a8.png" width=15% >

**Reply by binsee** [reply]
要打开『如果当前页面为适配站点，是否自动进入阅读模式？』这个开关

**Reply by Kenshin** [reply]
@okerivy 

请看 @binsee 的提示，位置在：选项页 → 高级设定 → 阅读模式

***

PS1：你截图中的仅为 **一级目录**，简悦支持 **开箱即用** & **灵活的定制化**，后者包含了超级多的定制化选项，所以左侧侧栏仅显示它们的 **模块** 功能而已。

PS2：新版会优化界面，使其更符合直觉。（包括此选项默认就可以看见）

**Comment by binsee** [comment]
适配效果已更新，现适配效果如下：

- 离线 HTML

  ![image](https://user-images.githubusercontent.com/5285894/120608457-cd5b3a80-c483-11eb-8ede-946a81e89d3d.png)

- Markdown

    <img width="777" alt="image" src="https://user-images.githubusercontent.com/81074/203249908-042ffd74-77c4-48ec-a16b-a0cb15c4a881.png">

----

当前能导出视频截图的，包括：

- 离线 HTML

- 离线 Markdown

- [assets + md](https://github.com/Kenshin/simpread/discussions/4041) 需要调整，[详细说明](https://github.com/Kenshin/simpread/issues/4510#issuecomment-1323116938)。

- [导出 PDF](https://github.com/Kenshin/simpread/discussions/3887)

- [Textbundle](https://simpread.pro/docs/#/Textbundle)

**Comment by daizy008** [comment]
虽然自己用不到。但是感觉超赞！忍不住过来回复下。
